python/cuml/test/dask/test_random_forest.py

```diff
@@ -68,34 +71,45 @@ def _prep_training_data(c, X_train, y_train, partitions_per_worker):
 
 
 @pytest.mark.parametrize('partitions_per_worker', [3])
-def test_rf_classification_dask_cudf(partitions_per_worker, client):
+def test_rf_classification_multi_class(partitions_per_worker, cluster):
 
     # Use CUDA_VISIBLE_DEVICES to control the number of workers
-    X, y = make_classification(n_samples=10000, n_features=20,
-                               n_clusters_per_class=1, n_informative=10,
-                               random_state=123, n_classes=5)
+    c = Client(cluster)
 
-    X = X.astype(np.float32)
-    y = y.astype(np.int32)
+    try:
 
-    X_train, X_test, y_train, y_test = \
-        train_test_split(X, y, test_size=1000)
+        X, y = make_classification(n_samples=10000, n_features=20,
+                                   n_clusters_per_class=1, n_informative=10,
+                                   random_state=123, n_classes=15)
 
-    cu_rf_params = {
-        'n_estimators': 40,
-        'max_depth': 16,
-        'n_bins': 16,
-    }
+        X = X.astype(np.float32)
+        y = y.astype(np.int32)
 
-    X_train_df, y_train_df = _prep_training_data(client, X_train, y_train,
-                                                 partitions_per_worker)
+        X_train, X_test, y_train, y_test = \
+            train_test_split(X, y, test_size=1000, random_state=123)
 
-    cuml_mod = cuRFC_mg(**cu_rf_params)
-    cuml_mod.fit(X_train_df, y_train_df)
-    cuml_mod_predict = cuml_mod.predict(X_test)
-    acc_score = accuracy_score(cuml_mod_predict, y_test, normalize=True)
+        cu_rf_params = {
+            'n_estimators': 25,
+            'max_depth': 16,
+            'n_bins': 16,
+        }
 
-    assert acc_score > 0.8
+        X_train_df, y_train_df = _prep_training_data(c, X_train, y_train,
+                                                     partitions_per_worker)
+
+        cuml_mod = cuRFC_mg(**cu_rf_params)
+        cuml_mod.fit(X_train_df, y_train_df)
+        cuml_preds_cpu = cuml_mod.predict(X_test, predict_model="CPU")
```

Comment:
should there be a comparison against sklearn or some other baseline (with some threshold)? Comparing FIL vs. CPU is useful for inference, but nothing then ensures that the training is correct.

Reply:
I went with approach of comparing the CPU vs GPU test results since the PR showcases FIL updates for multi classification. I thought that the comparison between cpu vs gpu will check if the FIL updates behave as expected for multi class classification in mnmg.
Furthermore, since we use large datasets and relatively large forests for rf mnmg tests it would take a long time to run sklearn models and it would increase the CI run time.

Reply:
I went with approach of comparing the CPU vs GPU test results since the PR showcases FIL updates for multi classification. I thought that the comparison between cpu vs gpu will check if the FIL updates behave as expected for multi class classification in mnmg.
Furthermore, since we use large datasets and relatively large forests for rf mnmg tests it would take a long time to run sklearn models and it would increase the CI run time.
With regards to the threshold, I am not sure on what would be a good approach to decide an acceptable threshold for RF mnmg multi class to be lower than sklearn's/cuml's single gpu RF accuracy.

Reply:
On running the test for cuml RF MNMG with sklearn as well I observed that after performing hyper parameter tuning the difference between cuml RF MNMG model and sklearn would be reduced to at most 0.45 (4.5%). Without hyper parameter tuning i have seen the difference in accuracy range from 0.7 to 0.5. The difference seems to get worse with an increase in n_estimators. The increase in cuml's accuracy is less than the increase in accuracy seen in sklearn's accuracy.